crypto/evp_extra/evp_asn1.c

```diff
@@ -91,18 +91,27 @@ static const EVP_PKEY_ASN1_METHOD *parse_key_type(CBS *cbs) {
   if (OBJ_cbs2nid(&oid) == NID_rsa) {
     return &rsa_asn1_meth;
   }
+
+#ifdef ENABLE_DILITHIUM
+  // if |cbs| is empty we overwrite the contents with |oid| so that we can
+  // call pub_decode(ret, &algorithm, &key) with the |algorithm| populated as |oid|.
+  // We could probably use CBS_peek_asn1_tag for this, to conditionally set |algorithm|
+  // based on if peeking at the next tag to see if there is params.
+  if (CBS_len(cbs) == 0) {
```

Comment:
Was this not needed for the private key?

Reply:
fixed my comment to call out its use for private keys in [c71bbac](https://github.com/aws/aws-lc/pull/1963/commits/c71bbac49160038ba8262b9f1996b43e197410ba)